src/main/java/seedu/address/model/Statistics.java

```diff
@@ -23,33 +28,120 @@ public Statistics(ReadOnlyStatistics toBeCopied) {
+        for (int i = 0; i < MAX_SIZE; i++) {
+            LocalDate tempLocalDate = currDate.minusDays(i);
+            String tempLocalDateStr = tempLocalDate.toString();
+            Date tempDate = new Date(tempLocalDateStr);
+            DayData dayData = new DayData(tempDate);
+            dayDataList.add(dayData);
+        }
+    }
+
     //// list overwrite operations
-    /**
-     * Replaces the contents of the person list with {@code persons}. {@code persons} must not
-     * contain duplicate persons.
-     */
+
+    /** Clears list */
+    public void clearList() {
+        this.dayDataList.clear();
+    }
+
+    /** Replaces the contents of the list with {@code dayDataList}. */
     public void setDayDatas(List<DayData> dayDataList) {
         this.dayDataList.clear();
         this.dayDataList.addAll(dayDataList);
     }
 
-    /** Resets the existing data of this {@code TaskList} with {@code newData}. */
+    /** Resets the existing data of this {@code Statistics} with {@code newData}. */
     public void resetData(ReadOnlyStatistics newData) {
         requireNonNull(newData);
         setDayDatas(newData.getDayDataList());
     }
 
-    //// person-level operations
+    //// dayData-level operations
+
+    // FUNCTIONS FOR
+    // HARDOHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHHH SIR
+
+    /** reinitialises dayDataList to current day while retaining stored data. */
+    public void fetch() {
+        HashMap<Date, PomDurationData> pomTempStorage = new HashMap<>();
+        HashMap<Date, TasksDoneData> tasksTempStorage = new HashMap<>();
+
+        for (int i = 0; i < dayDataList.size(); i++) {
+            pomTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getPomDurationData());
+            tasksTempStorage.put(
+                    dayDataList.get(i).getDate(), dayDataList.get(i).getTasksDoneData());
+        }
+
+        this.init();
+
+        for (int i = 0; i < dayDataList.size(); i++) { // updated dayDatalist
+            Date currTempDate = dayDataList.get(i).getDate();
+            if (pomTempStorage.containsKey(currTempDate)) {
+                PomDurationData tempPomDurationDate = pomTempStorage.get(currTempDate);
+                TasksDoneData tempTasksDoneData = tasksTempStorage.get(currTempDate);
+                DayData tempDayData =
+                        new DayData(currTempDate, tempPomDurationDate, tempTasksDoneData);
+                dayDataList.remove(i);
+                dayDataList.add(i, tempDayData);
+            }
+        }
+
+        assert (dayDataList.size() <= MAX_SIZE);
+    }
+
+    /**
+     * Auto-checks and replaces a new DayData object at the same Date
+     *
+     * @param dayData
+     */
+    public void updatesDayData(DayData dayData) {
+        requireNonNull(dayData);
+
+        Date currDate = dayData.getDate();
+        for (int i = 0; i < dayDataList.size(); i++) {
```

Comment:
Alright for an initial implementation, but is there a better way to update individual dates besides searching through all our DayData objects for the matching date?

Reply:
I considered updating just the head for the current day, but I realised that you will need to to update "yesterday"'s data in the event that pom goes past midnight. If we are using the queue structure suggested by Branson, probably not. But I think this is fine because we are storing a small dataset. Thoughts?

Reply:
That's true. Let's go with this for now!